### PR TITLE
Replace 'startsWith' with checking individual characters

### LINKED ISSUE #15558: String.startsWith() not supported by IE
Whilst investigating #15533 I found that IE11 does not support `String.protype.startsWith()` which is used [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/d9c166111b5d0b3d8e16c908e0f9dd73e5d6610a/common/app/routes/challenges/utils.js#L93) in our code base.

We need to refactor this method out

### PR DESCRIPTION
**Pull request by LukeBousfield**
## Pre-Submission Checklist

- [x] Branches much start with either `fix/` or `feature/`
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/         FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass `npm run test-challenges`.

## Description

Replaces `!value.startsWith('//')` with `!(value.charAt(0) === '/' && value.charAt(1) === '/')` to ensure Internet Explorer support.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Add new translation (feature adding new translations)

## Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #15558

### CODE REVIEW

**Review by ahstro — APPROVED** [review]

**Comment by CodingPower472** [comment]
When will this be merged?

**Comment by ahstro** [comment]
I don't have the ability to merge, so I don't know, sorry. 🙂

File: common/app/routes/challenges/utils.js [hunk 1]
```diff
@@ -90,7 +90,8 @@ function logReplacer(value) {
     const replaced = value.map(logReplacer);
     return '[' + replaced.join(', ') + ']';
   }
-  if (typeof value === 'string' && !value.startsWith('//')) {
+  if (typeof value === 'string' &&
+      !(value.charAt(0) === '/' && value.charAt(1) === '/')) {
```
Comment (Bouncey) [inline common/app/routes/challenges/utils.js]: I think a Regular Expression checking for `//` at the beginning will be a more elegant solution. It also keeps the number of checks required to a minimum

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Comment by Bouncey** [comment]
Thank you for providing a patch for this, please see my comment :+1:

**Comment by CodingPower472** [comment]
I'll change that. Thanks for the suggestion

**Comment by CodingPower472** [comment]
I switched to regex.

File: common/app/routes/challenges/utils.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ function logReplacer(value) {
     const replaced = value.map(logReplacer);
     return '[' + replaced.join(', ') + ']';
   }
-  if (typeof value === 'string' && !value.startsWith('//')) {
+  if (typeof value === 'string' && !RegExp('^//').test(value)) {
```
Comment (Bouncey) [inline common/app/routes/challenges/utils.js]: So, to use the constructor method of creating a `RegExp` we need the `new` keyword. This ensures it is has direct inheritance from `RegExp.prototype` and we are not leaving ourselves open to bugs or attacks. Also, it is better to escape the `/`'s for the same reasons.

You can just create a literal RegExp and test with 
```js
/^\/\//.test(value)

// -------or-------

const inlineCommentRE = /^\/\//;
inlineCommentRE.test(value)
```
which will sort of the inheritance for us. Note that the `/`'s are still escaped. 

Thanks for all your effort in this PR, I think one last change and it should be ready to merge ✨

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Comment by CodingPower472** [comment]
@Bouncey Done.

**Review by Bouncey — APPROVED** [review]

**Comment by Bouncey** [comment]
Great work @CodingPower472 thank you

## LGTM 👍

**Pull request merged**